src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -347,12 +341,12 @@ export function getInputType(
 
     function getInputModelType(m: Model): InputType {
         /* Array and Map Type. */
-        if (!isNeverType(m)) {
```

Comment:
we only get into this function when `m.kind === "Model"`, and the `isNeverType` is checking `m.kind === "intrinsic"` which will never be true, therefore here we do not need to check this.